### PR TITLE
Add Music Album and Genre Classes

### LINKED ISSUE #3: [0.5 pts] [TM2] Create MusicAlbum class
- [x] Create MusicAlbum class in a separate .rb file.

### LINKED ISSUE #4: [0.5 pts] [TM2] Add MusicAlbum class constructor
- [x] All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #6: [0.5 pts] [TM2] Create Genre class
- [x] Create Genre class with an association to the Item class in a separate .rb file.

### LINKED ISSUE #14: [0.5 pts] [TM2] Add Genre class constructor
- [x] All Genre class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #15: [1.0 pts] [TM2] Implement Genre class add_item method
Implement the `add_item` method in the Genre class with the following:

- [x] Should take an instance of the `Item` class as an input.
- [x] Should add the input item to the collection of items.
- [x] Should add `self` as a property of the `item` object (by using the correct setter from the item object).

### LINKED ISSUE #20: [1.0 pts] [TM2] Implement MusicAlbum can_be_archived?
Implement `MusicAlbum` `can_be_archived?` method with the following:

- [x] Should override the method from the parent class.
- [x] Should return true if parent's method returns true AND if on_spotify equals true.
- [x] Otherwise, it should return false

### LINKED ISSUE #24: [1.0 pts] [TM2] Implement App options for MusicAlbum and Genre
Implement the following options in the App options:

- [x] List all music albums.
- [x] List all genres (e.g 'Comedy', 'Thriller').
- [ ] Add a music album.

### PR DESCRIPTION
**Pull request by paulsaenzsucre**
## Add Music Album and Genre Classes

In this pull request I did the following:

📌 Create the `MusicAlbum` class with its `on_spotify` attribute, and its constructor and `can_be_archived?` methods.

📌 Create the `Genre` class with its `id`, `name` and `items` attributes, and its constructor and `add_item` methods.

📌 Implement the `list_genres`, `list_albums`, and `add_album` methods on `App` class.

📌 Implement the `seed_data` and `selected_genre` helper method on `App` class.

image

### CODE REVIEW

File: classes/genre.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Genre
+  attr_accessor :name
+
+  def initialize(name, id = SecureRandom.uuid)
```
Comment (Juank628) [inline classes/genre.rb:4]: I think the "id" shouldn't be an optional parameter. It always has to be generated by SecureRandom to avoid repeating the id.

**Review by Juank628 — CHANGES_REQUESTED** [review]

File: classes/genre.rb (hunk 1 above)
Comment (paulsaenzsucre) [reply classes/genre.rb:4]: Thank you for your comments. In this case I'm trying to get ahead of the saving of the data on files, I think we need a form to create the object with the id that comes from the JSON file. What do you think?

Comment (Juank628) [reply classes/genre.rb:4]: That is totally true! So, you can merge!

**Review by Juank628 — APPROVED** [review]
Excellent! You can merge

**Pull request merged**